test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go

```diff
@@ -118,4 +119,19 @@ func WaitForTuningUpdated(ctx context.Context, profile *performancev2.Performanc
 	Expect(err).ToNot(HaveOccurred())
 	err = nodepools.WaitForConfigToBeReady(ctx, testclient.ControlPlaneClient, np.Name, np.Namespace)
 	Expect(err).ToNot(HaveOccurred())
+
+	workerRTNodes, err := nodes.GetByLabels(testutils.NodeSelectorLabels)
+	if len(workerRTNodes) == 0 {
+		// Label the first node with worker-cnf label
+		nodesList := &corev1.NodeList{}
+		Expect(testclient.DataPlaneClient.List(context.TODO(), nodesList)).ToNot(HaveOccurred())
+		workerRTNode := nodesList.Items[0]
+		workerCNFLabel := fmt.Sprintf("%s/%s", testutils.LabelRole, testutils.RoleWorkerCNF)
+		testlog.Infof("labeling node %s with label %s\n", workerRTNode.Name, workerCNFLabel)
+		labels := workerRTNode.GetLabels()
+		labels[workerCNFLabel] = ""
+		workerRTNode.SetLabels(labels)
+		Expect(testclient.DataPlaneClient.Update(context.TODO(), &workerRTNode))
+		testlog.Infof("worker %s labels are: %v\n", workerRTNode.Name, workerRTNode.Labels)
+	}
```

Comment:
nope, the intention is good but this is a bit too simplistic.
1. a function which `wait` for something should not mutate the cluster state. Thus
2. let's move the label enforcement in its own function, and let's call it explicitely
3. inside the label updating logic, `get+mutate+update` needs to be wrapped into a `Eventually` or retry loop in general

Reply:
> 2. let's move the label enforcement in its own function, and let's call it explicitly

How do we avoid from other developers to forget to call this function?

Reply:
Done